paimon-core/src/main/java/org/apache/paimon/table/FallbackReadFileStoreTable.java

```diff
@@ -301,6 +302,15 @@ public List<BinaryRow> listPartitions() {
             partitions.addAll(fallbackScan.listPartitions());
             return new ArrayList<>(partitions);
         }
+
+        @Override
+        public List<PartitionEntry> listPartitionEntries() {
+            List<PartitionEntry> partitionEntries = mainScan.listPartitionEntries();
+            if (partitionEntries.isEmpty()) {
```

Comment:
if partition not exists in `mainScan`, add it into `partitionEntries`.

Please add test case too.